File: dep/frontend/src/serialization/node_desc.ts
import * as Graphflo from "graphflo/serialization";

export class NodeDesc {
    readonly title: string;
    readonly has_input: boolean;
    readonly has_output: boolean;
    readonly left_connectors: Graphflo.NodeConnectorDescs;
    readonly right_connectors: Graphflo.NodeConnectorDescs;
    readonly min_width?: number;
    readonly min_height?: number
}

File: dep/backend/src/serialization/node_desc.rs
use serialization::NodeConnectorDescs;

#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
pub struct NodeDesc {
    pub title: String,
    pub has_input: bool,
    pub has_output: bool,
    pub left_connectors: NodeConnectorDescs,
    pub right_connectors: NodeConnectorDescs,
    pub min_width: Option<u32>,
    pub min_height: Option<u32>,
}
